components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -525,6 +534,13 @@ protected boolean isAccountLock(String userName, UserStoreManager userStoreManag
             throw new AccountLockException("Error occurred while retrieving " + AccountConstants
                     .ACCOUNT_LOCKED_CLAIM + " claim value", e);
         }
+        if (isAccountLockBypassRoleExist(userStoreManager, userName)) {
```

Comment:
Better to not to do this inside isAccountLock method.

You can skip check that in the reference places like bellow

if (!isAccountLockBypassRoleExist(userStoreManager, userName)) {
isAccountLock();
}

Reply:
done